[Outlook] (preview) getItemIdAsync (#1051)
OfficeDev/office-js-docs-pr: docs/reference/objectmodel/preview-requirement-set/office.context.mailbox.item.md

---
---

#### getItemIdAsync([options], callback)

Asynchronously gets the ID of a saved item. Compose mode only.

When invoked, this method returns the item ID via the callback method.

> [!NOTE]

> > If your add-in calls `getItemIdAsync` on an item in compose mode in order to get an `itemId` to use with EWS or the REST API, be aware that when Outlook is in cached mode, it may take some time before the item is actually synced to the server. Until the item is synced, using the `itemId` will return an error.

Reviewer: ```suggestion
> If your add-in calls `getItemIdAsync` on an item in compose mode (e.g., to get an `itemId` to use with EWS or the REST API), be aware that when Outlook is in cached mode, it may take some time before the item is actually synced to the server. Until the item is synced, the `itemId` is not valid and using it will return an error.
```

##### Parameters

|Name|Type|Attributes|Description|
|---|---|---|---|
|`options`|Object|&lt;optional&gt;|An object literal that contains one or more of the following properties.|
|`options.asyncContext`|Object|&lt;optional&gt;|Developers can provide any object they wish to access in the callback method.|
|`callback`|function||When the method completes, the function passed in the `callback` parameter is called with a single parameter, `asyncResult`, which is an [`AsyncResult`](/javascript/api/office/office.asyncresult) object.<br/><br/>On success, the item identifier is provided in the `asyncResult.value` property.|

##### Errors

|Error code|Description|
|------------|-------------|

> |`ItemNotSaved`|The id can't be retrieved until the item is saved.|

Reviewer: ```suggestion
|`ItemNotSaved`|The ID can't be retrieved because the item has not yet been saved.|
```

Author: I believe this is the string that is actually returned as the error message.

##### Requirements

|Requirement|Value|
|---|---|
|[Minimum mailbox requirement set version](/office/dev/add-ins/reference/requirement-sets/outlook-api-requirement-sets)|Preview|
|[Minimum permission level](/outlook/add-ins/understanding-outlook-add-in-permissions)|ReadItem|
|[Applicable Outlook mode](/outlook/add-ins/#extension-points)|Compose|

##### Examples

```javascript
Office.context.mailbox.item.getItemIdAsync(
  function callback(result) {
    // Process the result.

>   });

Reviewer: ```suggestion
  }
);
```

```

> The following is an example of the `result` parameter passed to the callback function. The `value` property contains the item ID.

Reviewer: ```suggestion
The following example shows the structure of the `result` parameter that's passed to the callback function. The `value` property contains the item ID.
```
